lib/WUI/link_content/prusa_link_api.cpp

```diff
@@ -255,7 +254,12 @@ optional<ConnectionState> PrusaLinkApi::accept(const RequestParser &parser) cons
             return StatusPage(Status::MethodNotAllowed, StatusPage::CloseHandling::ErrorClose, parser.accepts_json);
         }
     } else if (suffix == "printer") {
-        return get_only(StatelessJson(get_printer, parser.can_keep_alive()));
+        return get_only(SendJson(EmptyRenderer(get_printer), parser.can_keep_alive()));
+    } else if (suffix == "download" || suffix == "transfer") {
+        if (transfers::Monitor::instance.status().has_value())
+            return get_only(SendJson(TransferRenderer(), parser.can_keep_alive()));
```

Comment:
I _think_ this is a little bit racy. Is it possible status has value now, but will disappear before entering the transfer renderer?

Let me have a look at that one 😇

Reply:
It is, maybe we could allow stale inside `TransferRenderer` every time, not depending on resume_point not being 0 and then we would be guaranteed to get a status.

Reply:
And maybe pass the expected transfer_id into the constructor, which could also eliminate a bit of repeated logic inside the render step.